app/services/estate_visit_query.rb

```diff
@@ -52,4 +69,14 @@ def grouped_visits(visits)
         end
       end
   end
+
+  def by_prisoner_number(number, visits)
+    number = Prisoner.normalise_number(number)
+    visits.joins(:prisoner).where(prisoners: { number: number }).to_a
+  end
+
+  def by_human_id(id, visits)
+    id = Prisoner.normalise_number(id)
```

Comment:
using the `normalise_number` method for the human_id makes sense given the implementation, however it looks out of place in this class. I think it should probably be implemented as a helper method used for searching.